atomix/cluster/src/main/java/io/atomix/raft/impl/ReconfigurationHelper.java

```diff
@@ -166,6 +170,170 @@ private void leaveInternal(final CompletableFuture<Void> future) {
+   *      |                                 |                                     |      Poll/Vote                     |
+   *      |                                 |                                     |<-----------------------------------|
+   *      |                                 |                                     |  Reject because log not uptodate   |
+   *      |                                 |                                     |----------------------------------->|
+   *      |                                 |                                     |                                    |
+   * </pre>
+   */
+  public CompletableFuture<Void> forceConfigure(final Collection<MemberId> newMembersIds) {
+    final CompletableFuture<Void> future = new CompletableFuture<>();
+
+    threadContext.execute(() -> triggerForceConfigure(newMembersIds, future));
+    return future;
+  }
+
+  private void triggerForceConfigure(
+      final Collection<MemberId> newMembersIds, final CompletableFuture<Void> future) {
+    final var currentConfiguration = raftContext.getCluster().getConfiguration();
+    final Collection<RaftMember> newMembers =
+        newMembersIds.stream()
+            // We can also take the type of the member as input. But so far we do not support
+            // PASSIVE members yet. So we can safely assume they are all ACTIVE.
+            .map(m -> new DefaultRaftMember(m, Type.ACTIVE, Instant.now()))
+            .collect(Collectors.toSet());
+
+    if (!currentConfiguration.force()) {
+      // No need to overwrite if it is already in force configure and this is a retry
+      if (raftContext.getRaftRole().role() == Role.LEADER) {
+        // Optimization: If the current configuration is already the same as new forced, we
+        // can skip reconfiguring. It is most likely a retry of a previous force request,
+        // which was interpreted as failure because of a request timeout.
+        raftContext.transition(Role.FOLLOWER);
+      }
+
+      logger.info(
+          "Current configuration is '{}'. Forcing configuration with members '{}'",
+          currentConfiguration,
+          newMembers);
+      final var newConfiguration =
+          new Configuration(
+              currentConfiguration.index() + 1,
+              raftContext.getTerm(),
+              Instant.now().toEpochMilli(),
+              newMembers,
+              Set.of(),
+              true);
+
+      raftContext.getCluster().configure(newConfiguration);
+    } else if (!(currentConfiguration.allMembers().containsAll(newMembers)
+        && newMembers.containsAll(currentConfiguration.allMembers()))) {
+      // This is not expected. When force configuration is retried, we expect that they are
+      // retried with the same state. If this is not the case, it is likely that there are two
+      // force configuration requested at the same time.
+      // Reject the request. There is possibly no way out to recover from this.
```

Comment:
:thought_balloon: That's a nice case to document, I assume you would need manual intervention for recovery? :thinking: Though our configuration file is not human editable =/

Reply:
I guess we can delete the raft configuration file and restart.